extensions/wikia/AdEngine/js/video/player/ui/panel.js

```diff
@@ -0,0 +1,45 @@
+/*global define, setTimeout*/
+define('ext.wikia.adEngine.video.player.ui.panel', [
+	'wikia.document',
+	'wikia.log'
+], function (doc, log) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.video.player.ui.panel';
+
+	function Panel(className, uiElements) {
+		var panelContainer;
+
+		if (!(uiElements instanceof Array)) {
+			log(['Panel', name, uiElements, 'uiElements must be an array'], 'error', logGroup);
+		}
+
+		panelContainer = doc.createElement('div')
+		panelContainer.className = className;
+		uiElements = uiElements.map(function (element) {
+			return 'ext.wikia.adEngine.video.player.ui.' + element;
+		});
+
+		this.getClassName = function () {
+			return className;
+		};
+
+		this.add = function (video, params) {
+			if (!params) {
+				params = {};
+			}
+
+			params.panelContainer = panelContainer;
```

Comment:
Imo it isn't safe. You're setting panelContainer in params "global" object, so it may be overridden by another Panel and the order of added theme components will be important

Reply:
uhm, it's global... damn. thanks